cmd/argocd-application-controller/commands/argocd_application_controller.go

```diff
@@ -67,22 +67,28 @@ func NewCommand() *cobra.Command {
 		Long:              "ArgoCD application controller is a Kubernetes controller that continuously monitors running applications and compares the current, live state against the desired target state (as specified in the repo). This command runs Application Controller in the foreground.  It can be configured by following options.",
 		DisableAutoGenTag: true,
 		RunE: func(c *cobra.Command, args []string) error {
+			vers := common.GetVersion()
+			namespace, _, err := clientConfig.Namespace()
+			errors.CheckError(err)
+			log.WithFields(log.Fields{
+				"version":   vers.Version,
+				"commit":    vers.GitCommit,
+				"built":     vers.BuildDate,
+				"namespace": namespace,
+			}).Info("ArgoCD Application Controller is starting")
```

Comment:
Is it okay that this happens before setting the log config below?

Reply:
Yes. In fact, it's deliberate so that this message appears even if info level logging is later disabled.